flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -471,15 +475,15 @@ public void setInitialState(@Nullable JobManagerTaskRestore taskRestore) {
                         });
     }
 
-    private static int getPartitionMaxParallelism(IntermediateResultPartition partition) {
-        final List<List<ExecutionEdge>> consumers = partition.getConsumers();
+    private static int getPartitionMaxParallelism(
```

Comment:
maybe `getPartitionMaxParallelism(IntermediateResultPartition, ExecutionGraph)` or `getPartitionMaxParallelism(IntermediateResultPartition, Function<ExecutionVertexID, ExecutionVertex>)`, so that we do not need to expose an `ExecutionGraph#getExecutionVertexMapping()`?